server.go

```diff
@@ -544,20 +604,26 @@ func (srv *Server) serveUDP(l *net.UDPConn) error {
 	rtimeout := srv.getReadTimeout()
 	// deadline is not used here
 	for {
-		m, s, err := reader.ReadUDP(l, rtimeout)
```

Comment:
What happens if you leave this line here?

Reply:
I need to have the dnsWg.Add before entering into the ReadUDP,
and I need it to be in the Lock (to ensure that srv.started is true when applying Add.

Reply:
I need that the Add happen BEFORE entering the ReadUDP.
and the Add needs to be checked against the srv.started ... 

So I moved the line below the block with Lock.

is there a specific concern ?

At this point of the workflow we do not know if there is a packet coming in or not : the ReadUDP is just waiting for next packet.
So I think that is safe to check the exit before the ReadUDP().

But it is NO MORE safe to test it between the ReadUDP and spanworker ... because we WANT that any coming msg is processed whatever srv.started.